### ISSUE TITLE
Why add dropout to attention score?

### ISSUE THREAD

**Issue by zqs01** [espnet/espnet#1502]
Hello everyone, I have a question. I use Transformer model for end-to-end ASR, Why add dropout to attention score? Any reference？ the path of attention.py is in espnet.nets.pytorch_backend.transformer.attention
[attention.py](https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/transformer/attention.py)

```        
scores = torch.matmul(q, k.transpose(-2, -1)) / math.sqrt(self.d_k)  # (batch, head, time1, time2)
        if mask is not None:
            mask = mask.unsqueeze(1).eq(0)  # (batch, 1, time1, time2)
            min_value = float(numpy.finfo(torch.tensor(0, dtype=scores.dtype).numpy().dtype).min)
            scores = scores.masked_fill(mask, min_value)
            self.attn = torch.softmax(scores, dim=-1).masked_fill(mask, 0.0)  # (batch, head, time1, time2)
        else:
            self.attn = torch.softmax(scores, dim=-1)  # (batch, head, time1, time2)
        p_attn = self.dropout(self.attn)
```
why use ` p_attn = self.dropout(self.attn)`? Any reference?

Thank you in advance.
